dts/bindings/gpio/microchip,mcp230xx.yaml

```diff
@@ -0,0 +1,28 @@
+#
+# Copyright (c) 2021 metraTec GmbH
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+description: |
+    This is a representation of the Microchip MCP230XX I2C Gpio Expander.
+
+compatible: "microchip,mcp230xx"
+
+include: [gpio-controller.yaml, i2c-device.yaml]
+
+properties:
+    label:
+      required: true
+
+    ngpios:
+      type: int
+      required: true
+      description: |
+        Number of gpios supported. To support both devices, the mcp23008 and the
+        mcp23017 this number is configurable. Possible values are <8> for the mcp23008
+        and <16> for the mcp23017.
```

Comment:
This looks like a candidate for `enum:` since only two values are supported. See https://docs.zephyrproject.org/latest/guides/dts/bindings.html#enum-values

Reply:
I believe we don't have a policy at the moment but we should probably have one .yaml per supported chip not per driver? I.e. we need separate .yaml for mcp23008 and mcp23017. `ngpios` property will have one value then and be defined either as `const: 16` or `const: 8`. This is not handled consistently in Zephyr at the moment.

Reply:
Is it possible to have multiple bindings per driver? Both, the mcp23008.yaml and
mcp23017.yaml would contain a `compatible: "microchip,mcp230xx"`. This is afaik not possible?

Reply:
Yes, it's possible to have multiple bindings per driver.

> Both, the mcp23008.yaml and mcp23017.yaml would contain a compatible: "microchip,mcp230xx"

Every .yaml would contain a different compatible.

At the bottom of the driver we would have something like
```
#undef DT_DRV_COMPAT
#define DT_DRV_COMPAT microchip_mcp23008
DT_INST_FOREACH_STATUS_OKAY(MCP230XX_INIT)

#undef DT_DRV_COMPAT
#define DT_DRV_COMPAT microchip_mcp23017
DT_INST_FOREACH_STATUS_OKAY(MCP230XX_INIT)
```
That would be my preferred approach since that's how Linux does it. But as I have mentioned - as far as I know - we don't have a Zephyr policy at the moment. A few more people, e.g. @galak @carlescufi @mbolivar-nordic, should express their opinion in the matter. Maybe we should discuss this shortly during the dev-review meeting.

Reply:
I am in favor of such an approach.